lib/ansible/modules/windows/win_xml.ps1

```diff
@@ -0,0 +1,194 @@
+#!powershell
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+Set-StrictMode -Version 2
+
+function Compare-XmlDocs($actual, $expected) {
+    if ($actual.Name -ne $expected.Name) {
+        throw "Actual name not same as expected: actual=" + $actual.Name + ", expected=" + $expected.Name
+    }
+    ##attributes...
+
+    if ($actual.Attributes -ne $null -and $expected.Attributes -ne $null) {
+        if ($actual.Attributes.Count -ne $expected.Attributes.Count) {
+           throw "attribute mismatch for actual=" + $actual.Name
+        }
+        for ($i=0;$i -lt $expected.Attributes.Count; $i =$i+1) {
+            if ($expected.Attributes[$i].Name -ne $actual.Attributes[$i].Name) {
+                throw "attribute name mismatch for actual=" + $actual.Name
+            }
+            if ($expected.Attributes[$i].Value -ne $actual.Attributes[$i].Value) {
+                throw "attribute value mismatch for actual=" + $actual.Name
+            }
+        }
+    }
+
+    ##children
+    if ($expected.ChildNodes.Count -ne $actual.ChildNodes.Count)  {
+        throw "child node mismatch. for actual=" + $actual.Name
+    }
+
+    for ($i=0;$i -lt $expected.ChildNodes.Count; $i =$i+1) {
+        if (-not $actual.ChildNodes[$i]) {
+            throw "actual missing child nodes. for actual=" + $actual.Name
+        }
+        Compare-XmlDocs $expected.ChildNodes[$i] $actual.ChildNodes[$i]
+    }
+
+    if ($expected.InnerText) {
+        if ($expected.InnerText -ne $actual.InnerText) {
+           throw "inner text mismatch for actual=" + $actual.Name
+        }
+    }
+    elseif ($actual.InnerText) {
+        throw "actual has inner text but expected does not for actual=" + $actual.Name
+    }
+}
+
+function BackupFile($path) {
+	$backuppath = $path + "." + [DateTime]::Now.ToString("yyyyMMdd-HHmmss");
+	Copy-Item $path $backuppath;
+	return $backuppath;
+}
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$debug_level = Get-AnsibleParam -obj $params -name "_ansible_verbosity" -type "int"
+$debug = $debug_level -gt 2
```

Comment:
The $debug variable is used to determine whether to return XML comparison errors. I can take all the functionality out or take the debug gate out. This was work in concert with removing the emission of the errors to a log on the remote server.